src/types/tweak_db_id.rs

```diff
@@ -0,0 +1,138 @@
+use std::fmt::Debug;
+use std::hash::Hash;
+
+use byteorder::{BigEndian, ByteOrder};
+use const_crc32::{crc32, crc32_seed};
+
+use crate::raw::root::RED4ext as red;
+
+#[derive(Default, Clone, Copy)]
+#[repr(transparent)]
+pub struct TweakDbId(red::TweakDBID);
+
+impl Debug for TweakDbId {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_tuple("TweakDbId")
+            .field(&unsafe { self.0.__bindgen_anon_1.value })
+            .finish()
+    }
+}
+
+impl PartialEq for TweakDbId {
+    fn eq(&self, other: &Self) -> bool {
+        unsafe { self.0.__bindgen_anon_1.value == other.0.__bindgen_anon_1.value }
+    }
+}
+
+impl Eq for TweakDbId {}
+
+impl PartialOrd for TweakDbId {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for TweakDbId {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        unsafe { self.0.__bindgen_anon_1.value }.cmp(&unsafe { other.0.__bindgen_anon_1.value })
```

Comment:
should also use either a helper or conversion
```suggestion
        u64::from(self).cmp(u64::from(other))
```

Reply:
done in 6a23d6d